### PR TITLE
[InstCombine] Fold an unsigned icmp of ucmp/scmp with a constant to an icmp of the original arguments

### PR DESCRIPTION
**Pull request by Poseydon42**
Proofs: https://alive2.llvm.org/ce/z/9mv8HU

The `ult` case in particular would resolve a regression caused by #101049

### CODE REVIEW

**Review by dtcxzyw — APPROVED** [review]
LGTM. Thank you!

File: llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp [hunk 1]
```diff
@@ -4018,6 +4018,16 @@ foldICmpOfCmpIntrinsicWithConstant(ICmpInst::Predicate Pred, IntrinsicInst *I,
       NewPredicate = ICmpInst::ICMP_ULE;
     break;
 
+  case ICmpInst::ICMP_ULT:
+    if (C.ugt(1) && !C.isAllOnes())
```
Comment (nikic) [inline llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp:4022]:
```suggestion
    if (C.ugt(1))
```
This check doesn't appear to be necessary: https://alive2.llvm.org/ce/z/NyMjpe

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp (hunk 1 above)
Comment (Poseydon42) [reply llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp]: Yep, fixed.

**Pull request merged**